custom_components/flexmeasures_hacs/config_flow.py

```diff
@@ -1,58 +1,88 @@
+            default="example@example.com",
         ): str,
-        vol.Required(
-            "username", description={"suggested_value": "toy-user@flexmeasures.io"}
-        ): str,
-        vol.Required("password"): str,
-        vol.Required("power_sensor", description={"suggested_value": 1}): int,
-        vol.Required(
+        vol.Required("password", default="password"): str,
+        vol.Optional("power_sensor", default=5): int,
+        vol.Optional("soc_minima_sensor_id", default=5): int,
+        vol.Optional("soc_maxima_sensor_id", default=5): int,
+        vol.Optional("fill_level_sensor_id", default=5): int,
+        vol.Optional("fill_rate_sensor_id", default=5): int,
+        vol.Optional("usage_forecast_sensor_id", default=5): int,
+        vol.Optional("thp_fill_rate_sensor_id", default=5): int,
+        vol.Optional("thp_efficiency_sensor_id", default=5): int,
+        vol.Optional("nes_fill_rate_sensor_id", default=5): int,
+        vol.Optional("nes_efficiency_sensor_id", default=5): int,
+        vol.Optional("rm_discharge_sensor_id", default=5): int,
+        vol.Optional("schedule_duration", default="PT24H"): str,
+        vol.Optional(
             "consumption_price_sensor", description={"suggested_value": 2}
         ): int,
-        vol.Required(
+        vol.Optional(
             "production_price_sensor", description={"suggested_value": 2}
         ): int,
-        vol.Required("soc_sensor", description={"suggested_value": 4}): int,
-        vol.Required("rm_discharge_sensor", description={"suggested_value": 5}): int,
-        vol.Required(
-            "schedule_duration", description={"suggested_value": "PT24H"}
-        ): str,
-        vol.Required("soc_unit", description={"suggested_value": "kWh"}): str,
-        vol.Required("soc_min", description={"suggested_value": 0.001}): float,
-        vol.Required("soc_max", description={"suggested_value": 0.002}): float,
+        vol.Optional("soc_sensor", description={"suggested_value": 4}): int,
+        vol.Optional("soc_unit", default="kWh"): str,
+        vol.Optional("soc_min", default=10.1): float,
+        vol.Optional("soc_max", default=1.1): float,
     }
 )
 
 
-async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+def get_host_and_ssl_from_url(url: str) -> tuple[str, bool]:
+    """Get the host and ssl from the url."""
+    if url.startswith("http://"):
+        ssl = False
+        host = url.removeprefix("http://")
+    elif url.startswith("https://"):
+        ssl = True
+        host = url.removeprefix("https://")
+    else:
+        ssl = True
+        host = url
+
+    return host, ssl
+
+
+async def validate_input(
+    handler: SchemaCommonFlowHandler, data: dict[str, Any]
+) -> dict:
     """Validate if the user input allows us to connect.
 
-    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    Data has the keys from CONFIG_SCHEMA with values provided by the user.
```

Comment:
`CONFIG_SCHEMA` or `SCHEMA`? In case we will have multiple schema it might be handy to already start naming them distinctly. For instance, might we need an `OPTIONS_SCHEMA`, too, at some point?